improved test r2 to 0.83 and rmse to 33646

josepmariall opened the pull request:
Hello Soledad, 

I managed to improved the model.

Sincerely,

Josep Maria

titoniubo-tech committed:
improved test r2 to 0.83 and rmse to 33646

solegalli commented on 13-Assignement.ipynb:
```diff
@@ -76,13 +86,26 @@
+    "\n",
+    "# drop variables with more than 80% of missing data\n",
+    "# namely \"Alley\", \"Fence\", \"MiscFeature\", \"PoolQC\"\n",
+    "data.drop(['Alley', 'Fence', 'MiscFeature','PoolQC'], axis=1, inplace=True)\n"
```
Hi Josep, one of the rules was not to "select" features, which also implies not to remove features :)

solegalli commented on 13-Assignement.ipynb:
```diff
@@ -76,13 +86,26 @@
     "# load dataset\n",
     "\n",
-    "data = pd.read_csv('../houseprice.csv')"
+    "data = pd.read_csv('houseprice.csv')"
```
Could you design the notebook so that it reads the data as ../houseprice.csv?
This way, it looks like what other students are used to, and I would upload your notebook to replace mine.

solegalli reviewed (changes requested):
Hi Josep, well done. That was a massive improvement. 

Would you be able to re-do the model without deleting those features because they had a lot of NA?
In real like one would do as you did, but for the purpose of this challenge, I created this rule so that we measure only the feature engineering steps.

Also, would you be able to rename the jupyter notebook by adding your name or username? so that your solution is available for other students to learn from?

Also, did you try submitting our result to kaggle? how did it go?

Thank you
Sole

josepmariall commented:
Hello Soledad,

Many thanks for your feedback, I really appreciate it.

You are right about the submission. Reading the rules again I have just
realised I was not allowed to do feature selection. My apologies.

I shall modify my submission as per your request and do a PR again.

As for Kaggle, I appreciate your encouragement. I have never submitted
anything to Kaggle, I shall try and let you know.

Have a nice day.

Sincerely,

Josep Maria Niubo

On Thu, Oct 24, 2019 at 11:04 AM Soledad Galli <notifications@github.com>
wrote:

> *@solegalli* requested changes on this pull request.
>
> Hi Josep, well done. That was a massive improvement. Would you be able to
> re-do the model without deleting those features because they had a lot of
> NA?
> In real like one would do as you did, but for the purpose of this
> challenge, I created this rule so that we measure only the feature
> engineering steps.
>
> If you were able to re do the model, and attend the other 2 comments, I
> would use your notebook as the new challenge for the course.
>
> Also, did you try submitting our result to kaggle? how did it go?
>
> Thank you
> Sole
> ------------------------------
>
> In 13-Assignement.ipynb
> <https://github.com/solegalli/udemy-feml-challenge/pull/2#discussion_r338459122>
> :
>
> >     ]
>    },
>    {
>     "cell_type": "code",
>     "execution_count": 3,
>     "metadata": {},
> +   "outputs": [],
> +   "source": [
> +    "# Drop columsn with too many NA (over 80%)\n",
> +    "\n",
> +    "# drop variables with more than 80% of missing data\n",
> +    "# namely \"Alley\", \"Fence\", \"MiscFeature\", \"PoolQC\"\n",
> +    "data.drop(['Alley', 'Fence', 'MiscFeature','PoolQC'], axis=1, inplace=True)\n"
>
> Hi Josep, one of the rules was not to "select" features, which also
> implies not to remove features :)
> ------------------------------
>
> In 13-Assignement.ipynb
> <https://github.com/solegalli/udemy-feml-challenge/pull/2#discussion_r338459593>
> :
>
> > @@ -76,13 +86,26 @@
>     "source": [
>      "# load dataset\n",
>      "\n",
> -    "data = pd.read_csv('../houseprice.csv')"
> +    "data = pd.read_csv('houseprice.csv')"
>
> Could you design the notebook so that it reads the data as
> ../houseprice.csv?
> This way, it looks like what other students are used to, and I would
> upload your notebook to replace mine.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/solegalli/udemy-feml-challenge/pull/2?email_source=notifications&email_token=AJTVBSXBEPCEKNQXDFFGFXDQQFQKLA5CNFSM4JEEENO2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCJBZPAA#pullrequestreview-306419584>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJTVBSSKYHW2HL2HROCDUMLQQFQKLANCNFSM4JEEENOQ>
> .
>

titoniubo-tech committed:
modified as per request

Now complying with rules, no feature selection. Also file location modified as per request

titoniubo-tech committed:
add git ignore

solegalli commented:
Hi Josep, awesome.

You don't need to do a new PR, you just need to commit your new changes to this same branch your are using and it will update automatically.

Having said this, I had to change the repo substantially after encountering a few issues with the first solution submissions, so for this time and just in case, make a PR and email me your notebook, so I can upload it myself in case the PR raises merge issues.

Thank you

Sole

titoniubo-tech committed:
updated (removed unnecessary line)
